### PR TITLE
Signup: load fonts in `SiteMockup`

### PR DESCRIPTION
**Pull request by mattwiebe**
The font needed for any given site style will be loaded and rendered. Fonts are stored in the site styles config.

Before: 
image

After:
image

### Changes proposed in this Pull Request

* Load and apply Google fonts for site styles

### Testing instructions

1. Go to `http://calypso.localhost:3000/start/onboarding-dev/`
2. Note the serif font (Crimson Text) in the `SiteMockup` components, where before it was just the Calypso font.

The font should only ever be loaded once. Verify by inspecting the end of the `<head>` element.

### CODE REVIEW

File: client/lib/signup/site-styles.js [hunk 1]
```diff
@@ -19,6 +19,11 @@ export const siteStyleOptions = {
 			id: 'default',
 			label: 'Radcliffe Perfect',
 			theme: 'pub/radcliffe-2',
+			font: {
+				name: 'Crimson Text',
+				// variations in fvd format: https://github.com/typekit/fvd
+				variations: [ 'n4', 'n7' ],
```
Comment (ramonjd) [inline client/lib/signup/site-styles.js]: 👍

File: client/lib/signup/site-styles.js [hunk 2]
```diff
@@ -19,6 +19,11 @@ export const siteStyleOptions = {
 			id: 'default',
 			label: 'Radcliffe Perfect',
 			theme: 'pub/radcliffe-2',
+			font: {
```
[the remark this answers is not in the document]
Comment (fditrapani) [reply client/lib/signup/site-styles.js]: We definitely want to avoid the flash effect.

[the remark this answers is not in the document]
Comment (mattwiebe) [reply client/lib/signup/site-styles.js]: We have always been at war with the Flash of Unstyled Text. There will be no perfect solution to this but I'll see if we can solve the worst of it.

[the remark this answers is not in the document]
Comment (shaunandrews) [reply client/lib/signup/site-styles.js]: It's very noticeable, even when loading the default font.

**Comment by shaunandrews** [comment]
Outside of the Flash of Unstyled Text ™, this seems to work as expected, and loads in both 400 and 700 weights. 💯

**Comment by fditrapani** [comment]
> Hide text until the font is loaded

I'm not sure if this looks better than before. Any chance we can preload the fonts earlier in the flow?

**Comment by michaeldcain** [comment]
@fditrapani: can this alternatively be solved by some sort of transition animation between styles?

**Comment by fditrapani** [comment]
> can this alternatively be solved by some sort of transition animation between styles?

@michaeldcain I was thinking that as another approach as well but didn't have any transitions in mind that would feel like they get in the way. That being said, it's a viable option too.

**Comment by shaunandrews** [comment]
Looks good enough to me. Nice work around, @mattwiebe.

File: client/signup/site-mockup/index.jsx [hunk 3]
```diff
@@ -43,6 +44,31 @@ class SiteMockups extends Component {
...
 
+	constructor( props ) {
+		super( props );
+		this.state = this.getFontLoaderState( props );
+	}
+
+	getFontLoaderState( props ) {
+		const state = {
+			fontLoaded: false,
+			fontLoader: loadFont( props.siteStyle, props.siteType ),
```
Comment (ramonjd) [inline client/signup/site-mockup/index.jsx]: Is `this.state.fontLoader()` used anywhere? Not a biggie, but could we just `loadFont` straight up without assignment? 

```
		loadFont( props.siteStyle, props.siteType )
			.then( () => this.setState( { fontLoaded: true } ) );
```

**Comment by fditrapani** [comment]
> I think it's a great start. LGTM contingent upon a design decision re: flashing, however I don't think it's a real blocker since we can make another pass before this feature goes live.

I'd be ok with moving this along and dealing with the flash separately.  Thanks for the suggestion @ramonjd 👊 


also cc @shaunandrews to get his thoughts on the flash. I tried this out locally vs looking at the gif and it actually worked really well with my fast connection. I worry for folks that have slower connections though.

File: client/signup/site-mockup/index.jsx (hunk 3 above)
Comment (mattwiebe) [reply client/signup/site-mockup/index.jsx]: Yeah, it's a local state but not in any way tied to rendering so it shouldn't really be on `this.state`. Good catch.

**Comment by mattwiebe** [comment]
The flash was because it turns out I was only preloading the regular weight of the new font, so you were seeing a flash as the bold weight for the titles was being downloaded, parsed, and swapped in.

The latest change preloads all of the variations that are specified and now it's pretty dang near perfect.

It also now properly handles an error state where the fonts can't be downloaded for whatever reason. The previews will just be rendered in Calypso's font, then.

**Comment by fditrapani** [comment]
Looks sharp. Thanks for the updates @mattwiebe!

**Pull request merged**